kernel/intel_ethernet/src/descriptors.rs

```diff
@@ -320,4 +326,50 @@ impl fmt::Debug for AdvancedRxDescriptor {
 }
 
 
+/// Advanced Transmit Descriptor used in the Ixgbe driver.
+/// It has 2 modes: Read and Write Back, both of which use the whole 128 bits. 
+/// There is one transmit descriptor per transmit buffer that can be converted between these 2 modes.
+/// Read contains the addresses that the driver writes.
+/// Write Back contains information the hardware writes on receiving a packet.
+/// More information can be found in the 82599 datasheet.
+#[derive(FromBytes)]
+#[repr(C)]
+pub struct AdvancedTxDescriptor {
+    /// Starting physcal address of the receive buffer for the packet.
+    pub packet_buffer_address:  Volatile<u64>,
+    /// Length of data buffer
+    pub data_len: Volatile<u16>,
+    /// dtyp = Descriptor Type
+    /// mac = option to apply LinkSec and time stamp
```

Comment:
Is this field two items combined into one `u8`? If so, it might be good to define which bits are used by whom, e.g.,
```
dtyp occupied bits [0:4), mac occupies bits [4:8)
```
You could also use [0:3] and [4:7] if you prefer inclusive ranges

Reply:
Another way to do it is to use the bitfield type, but I dunno if that would interfere with how this field is actually used in the code. Take a look at it though.